<?xml version="1.0" encoding="UTF-8"?>
<testsuite name="org.apache.kafka.common.security.ssl.SslFactoryTest" tests="21" skipped="0" failures="0" errors="0" timestamp="2020-10-29T02:32:19" hostname="chukun-mac" time="11.892">
  <properties/>
  <testcase name="testKeyStoreTrustStoreValidation[tlsProtocol=TLSv1.2]" classname="org.apache.kafka.common.security.ssl.SslFactoryTest" time="2.604"/>
  <testcase name="testReconfiguration[tlsProtocol=TLSv1.2]" classname="org.apache.kafka.common.security.ssl.SslFactoryTest" time="0.637"/>
  <testcase name="testReconfigurationWithoutTruststore[tlsProtocol=TLSv1.2]" classname="org.apache.kafka.common.security.ssl.SslFactoryTest" time="0.799"/>
  <testcase name="testSslFactoryWithoutPasswordConfiguration[tlsProtocol=TLSv1.2]" classname="org.apache.kafka.common.security.ssl.SslFactoryTest" time="0.276"/>
  <testcase name="testReconfigurationWithoutKeystore[tlsProtocol=TLSv1.2]" classname="org.apache.kafka.common.security.ssl.SslFactoryTest" time="0.666"/>
  <testcase name="testKeystoreVerifiableUsingTruststore[tlsProtocol=TLSv1.2]" classname="org.apache.kafka.common.security.ssl.SslFactoryTest" time="1.042"/>
  <testcase name="staleSslEngineFactoryShouldBeClosed[tlsProtocol=TLSv1.2]" classname="org.apache.kafka.common.security.ssl.SslFactoryTest" time="0.668"/>
  <testcase name="testServerSpecifiedSslEngineFactoryUsed[tlsProtocol=TLSv1.2]" classname="org.apache.kafka.common.security.ssl.SslFactoryTest" time="0.551"/>
  <testcase name="testPemReconfiguration[tlsProtocol=TLSv1.2]" classname="org.apache.kafka.common.security.ssl.SslFactoryTest" time="0.935"/>
  <testcase name="testUntrustedKeyStoreValidationFails[tlsProtocol=TLSv1.2]" classname="org.apache.kafka.common.security.ssl.SslFactoryTest" time="0.345"/>
  <testcase name="testSslFactoryWithIncorrectProviderClassConfiguration[tlsProtocol=TLSv1.2]" classname="org.apache.kafka.common.security.ssl.SslFactoryTest" time="0.015"/>
  <testcase name="testCertificateEntriesValidation[tlsProtocol=TLSv1.2]" classname="org.apache.kafka.common.security.ssl.SslFactoryTest" time="0.776"/>
  <testcase name="testInvalidSslEngineFactory[tlsProtocol=TLSv1.2]" classname="org.apache.kafka.common.security.ssl.SslFactoryTest" time="0.002"/>
  <testcase name="testSslFactoryWithCustomKeyManagerConfiguration[tlsProtocol=TLSv1.2]" classname="org.apache.kafka.common.security.ssl.SslFactoryTest" time="0.379"/>
  <testcase name="testClientMode[tlsProtocol=TLSv1.2]" classname="org.apache.kafka.common.security.ssl.SslFactoryTest" time="0.003"/>
  <testcase name="testSslFactoryConfiguration[tlsProtocol=TLSv1.2]" classname="org.apache.kafka.common.security.ssl.SslFactoryTest" time="0.269"/>
  <testcase name="testClientSpecifiedSslEngineFactoryUsed[tlsProtocol=TLSv1.2]" classname="org.apache.kafka.common.security.ssl.SslFactoryTest" time="0.003"/>
  <testcase name="testEngineFactoryClosed[tlsProtocol=TLSv1.2]" classname="org.apache.kafka.common.security.ssl.SslFactoryTest" time="0.002"/>
  <testcase name="testPemKeystoreVerifiableUsingTruststore[tlsProtocol=TLSv1.2]" classname="org.apache.kafka.common.security.ssl.SslFactoryTest" time="0.989"/>
  <testcase name="testPemCertificateEntriesValidation[tlsProtocol=TLSv1.2]" classname="org.apache.kafka.common.security.ssl.SslFactoryTest" time="0.923"/>
  <testcase name="testSslFactoryWithoutProviderClassConfiguration[tlsProtocol=TLSv1.2]" classname="org.apache.kafka.common.security.ssl.SslFactoryTest" time="0.001"/>
  <system-out><![CDATA[[2020-10-29 10:32:21,950] ERROR Reclaimed buffer of size 95446630 and identity 764737691 that was not properly release()ed. This is a bug. (org.apache.kafka.common.memory.GarbageCollectedMemoryPool:106)
[2020-10-29 10:32:21,950] ERROR Reclaimed buffer of size 10 and identity 1564199829 that was not properly release()ed. This is a bug. (org.apache.kafka.common.memory.GarbageCollectedMemoryPool:106)
[2020-10-29 10:32:21,951] ERROR Reclaimed buffer of size 95446630 and identity 1552539841 that was not properly release()ed. This is a bug. (org.apache.kafka.common.memory.GarbageCollectedMemoryPool:106)
[2020-10-29 10:32:21,951] ERROR Reclaimed buffer of size 10 and identity 1985817918 that was not properly release()ed. This is a bug. (org.apache.kafka.common.memory.GarbageCollectedMemoryPool:106)
[2020-10-29 10:32:21,954] ERROR Reclaimed buffer of size 10 and identity 1523532126 that was not properly release()ed. This is a bug. (org.apache.kafka.common.memory.GarbageCollectedMemoryPool:106)
[2020-10-29 10:32:21,952] ERROR Reclaimed buffer of size 95446630 and identity 2093077239 that was not properly release()ed. This is a bug. (org.apache.kafka.common.memory.GarbageCollectedMemoryPool:106)
[2020-10-29 10:32:21,955] ERROR Reclaimed buffer of size 95446630 and identity 264712385 that was not properly release()ed. This is a bug. (org.apache.kafka.common.memory.GarbageCollectedMemoryPool:106)
[2020-10-29 10:32:21,955] ERROR Reclaimed buffer of size 95446630 and identity 154532707 that was not properly release()ed. This is a bug. (org.apache.kafka.common.memory.GarbageCollectedMemoryPool:106)
[2020-10-29 10:32:21,955] ERROR Reclaimed buffer of size 95446630 and identity 2061224905 that was not properly release()ed. This is a bug. (org.apache.kafka.common.memory.GarbageCollectedMemoryPool:106)
[2020-10-29 10:32:21,955] ERROR Reclaimed buffer of size 95446630 and identity 467894814 that was not properly release()ed. This is a bug. (org.apache.kafka.common.memory.GarbageCollectedMemoryPool:106)
[2020-10-29 10:32:21,955] ERROR Reclaimed buffer of size 95446630 and identity 1774695355 that was not properly release()ed. This is a bug. (org.apache.kafka.common.memory.GarbageCollectedMemoryPool:106)
[2020-10-29 10:32:23,066] ERROR Modification time of key store could not be obtained: /var/folders/y3/mvtqxgns6j93y74hb90_t2p9rflg_z/T/serverKS4515314352738123302.jks (org.apache.kafka.common.security.ssl.DefaultSslEngineFactory:385)
java.nio.file.NoSuchFileException: /var/folders/y3/mvtqxgns6j93y74hb90_t2p9rflg_z/T/serverKS4515314352738123302.jks
	at sun.nio.fs.UnixException.translateToIOException(UnixException.java:86)
	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:102)
	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:107)
	at sun.nio.fs.UnixFileAttributeViews$Basic.readAttributes(UnixFileAttributeViews.java:55)
	at sun.nio.fs.UnixFileSystemProvider.readAttributes(UnixFileSystemProvider.java:144)
	at java.nio.file.Files.readAttributes(Files.java:1737)
	at java.nio.file.Files.getLastModifiedTime(Files.java:2266)
	at org.apache.kafka.common.security.ssl.DefaultSslEngineFactory$FileBasedStore.lastModifiedMs(DefaultSslEngineFactory.java:383)
	at org.apache.kafka.common.security.ssl.DefaultSslEngineFactory$FileBasedStore.modified(DefaultSslEngineFactory.java:391)
	at org.apache.kafka.common.security.ssl.DefaultSslEngineFactory.shouldBeRebuilt(DefaultSslEngineFactory.java:109)
	at org.apache.kafka.common.security.ssl.SslFactory.createNewSslEngineFactory(SslFactory.java:150)
	at org.apache.kafka.common.security.ssl.SslFactory.reconfigure(SslFactory.java:117)
	at org.apache.kafka.common.security.ssl.SslFactoryTest.testReconfiguration(SslFactoryTest.java:247)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.runTestClass(JUnitTestClassExecutor.java:110)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.execute(JUnitTestClassExecutor.java:58)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.execute(JUnitTestClassExecutor.java:38)
	at org.gradle.api.internal.tasks.testing.junit.AbstractJUnitTestClassProcessor.processTestClass(AbstractJUnitTestClassProcessor.java:62)
	at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.processTestClass(SuiteTestClassProcessor.java:51)
	at sun.reflect.GeneratedMethodAccessor35.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:33)
	at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:94)
	at com.sun.proxy.$Proxy2.processTestClass(Unknown Source)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.processTestClass(TestWorker.java:119)
	at sun.reflect.GeneratedMethodAccessor34.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:182)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:164)
	at org.gradle.internal.remote.internal.hub.MessageHub$Handler.run(MessageHub.java:414)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
	at java.lang.Thread.run(Thread.java:748)
[2020-10-29 10:32:28,361] ERROR Unrecognized security provider creator class (org.apache.kafka.common.config.SecurityConfig:94)
java.lang.ClassNotFoundException: com.fake.ProviderClass1
	at java.net.URLClassLoader.findClass(URLClassLoader.java:382)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:349)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at java.lang.Class.forName0(Native Method)
	at java.lang.Class.forName(Class.java:264)
	at org.apache.kafka.common.utils.SecurityUtils.addConfiguredSecurityProviders(SecurityUtils.java:86)
	at org.apache.kafka.common.security.ssl.DefaultSslEngineFactory.configure(DefaultSslEngineFactory.java:136)
	at org.apache.kafka.common.security.ssl.SslFactory.instantiateSslEngineFactory(SslFactory.java:136)
	at org.apache.kafka.common.security.ssl.SslFactory.configure(SslFactory.java:93)
	at org.apache.kafka.common.security.ssl.SslFactoryTest.testSslFactoryWithIncorrectProviderClassConfiguration(SslFactoryTest.java:135)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.ExpectException.evaluate(ExpectException.java:19)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.runTestClass(JUnitTestClassExecutor.java:110)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.execute(JUnitTestClassExecutor.java:58)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.execute(JUnitTestClassExecutor.java:38)
	at org.gradle.api.internal.tasks.testing.junit.AbstractJUnitTestClassProcessor.processTestClass(AbstractJUnitTestClassProcessor.java:62)
	at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.processTestClass(SuiteTestClassProcessor.java:51)
	at sun.reflect.GeneratedMethodAccessor35.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:33)
	at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:94)
	at com.sun.proxy.$Proxy2.processTestClass(Unknown Source)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.processTestClass(TestWorker.java:119)
	at sun.reflect.GeneratedMethodAccessor34.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:182)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:164)
	at org.gradle.internal.remote.internal.hub.MessageHub$Handler.run(MessageHub.java:414)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
	at java.lang.Thread.run(Thread.java:748)
]]></system-out>
  <system-err><![CDATA[]]></system-err>
</testsuite>
